python/graphstorm/gconstruct/file_io.py

```diff
@@ -131,9 +141,13 @@ def read_data_json(data_file, data_fields):
         for key in data_fields:
             assert key in record, \
                     f"The data field {key} does not exist in the record {record} of {data_file}."
-            data[key].append(record[key])
+            record1 = np.array(record[key]) if isinstance(record[key], list) else record[key]
+            data[key].append(record1)
     for key in data:
-        data[key] = np.array(data[key])
+        if isinstance(data[key][0], np.ndarray):
```

Comment:
Is it possible that Process 0 has max_len 10 while Process 1 has max_len 12. for example.

Reply:
this is a concern. if this happens, we should return an error.

Reply:
I'll submit another PR to handle errors in the worker processes.